components/Init/src/main.c

```diff
@@ -174,12 +173,12 @@ void pre_init(void)
         ZF_LOGF("Failed to allocate reservation");
     }
     error = sel4utils_reserve_range_no_alloc(&vspace, pool_reservation.res,
```

Comment:
I think with the current parameters we should have this here. And we can't refer to `ALLOCMAN_VIRTUAL_SIZE` any longer because this got removed, so I adjusted the comment.
```
* For every 500.000.000 bytes in the allocator virtual size
 * the guest can map 6.960 MiBs. Dividing the two
 * values gives a scale of 71839, which we round up to 72000
 */
unsigned in = allocman_virt_size = get_guest_ram_mb() * 72000;
```
My point is, that the template does not really know what we do here and it should not have to (at least for now). Your current improvement goal seem to be making things more dynamic, but we should separate this from aim int towards  for config option about for the allocator pool (like we have on ARM). So I consider it more simple for now if we just provide a `get_guest_ram_mb()`. As a follow-up we could consider aligning the x86 and ARM a bit more then

Reply:
Its not trying to make things more dynamic really, its just trying to allow for large RAM allocation, replacing a predefined macro with a value calculated based on the VMs needs. ARM doesn't define the amount of RAM in the configuration, x86 does, and its used in other places throughout the VMM.